### PR TITLE
翻译完成了README的全文

### PR DESCRIPTION
**Pull request by wjsvec**
可能有一些英语不地道的问题，语法错误已经排查过了应该没有什么问题~

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
hi @AbdelouahabMbarki, please feel free to help us to review this translation (from Chinese to English). I know that you might not be familiar with Chinese. So, you might be able to focus on the English grammar part.

Even better, if you're interested in translating it to your first language. Thanks for your time to read this comment.

File: README-en.md [hunk 1]
```diff
@@ -6,134 +6,141 @@
...
+  * We can use [YouTube](https://www.youtube.com/) for video records.
+
+## Others
+Here are some other recommended habits and methods:
+* Checking email habitually
+  * Please be sure to associate your usual email address with GitHub, and then you can receive the notice information in issues and PR in time.
+* Manage your schedule through your computer and mobile phone
+  * There will be various meetings in the open-source community, and it's easy to forget important things if you're not in the habit of using a calendar.
+
+## Other references
```
Comment (LinuxSuRen) [inline README-en.md:142]: How about just using `References`? Of course, the Chinese could be `参考资料`

**Review by LinuxSuRen — COMMENTED** [review]
Thanks for your awesome work. I left a small comment.

**Comment by AbdelouahabMbarki** [comment]
> hi @AbdelouahabMbarki, please feel free to help us to review this translation (from Chinese to English). I know that you might not be familiar with Chinese. So, you might be able to focus on the English grammar part.
> 
> Even better, if you're interested in translating it to your first language. Thanks for your time to read this comment.

I can do a full review of the English version, btw, I think the project can be simpler if we have the same project structure as this  [one](https://github.com/kamranahmedse/developer-roadmap)

**Comment by LinuxSuRen** [comment]
kamranahmedse/developer-roadmap is a pretty interesting project.

**Comment by AbdelouahabMbarki** [comment]
I found this two repositories:
- https://github.com/github/opensource.guide
- https://github.com/freeCodeCamp/how-to-contribute-to-open-source

File: README-en.md [hunk 2]
```diff
@@ -6,134 +6,141 @@
...
-  * 如果没有相关的 team 可以 cc 的话，可以找最近合并过类似 PR 的人，并说明是由于找不到其他的方式，以及表示抱歉打扰
+Common Misunderstanding:
+* Use a single branch(like master) to commit changes.
+* Many different updates and fixing in a single PR.
+* Continually add new content in a single PR.
+* Merge your own PR.
+* Use instant messaging to urge specific people to review your PR.
+
+Best Practice:
+* Before submitting PR for the first time, browse some successfully merged PR comments and wanted formatted, etc.
```
Comment (Felixnoo) [inline README-en.md:51]: `merged PR comments and wanted formatted` -> `merged PRs and take a look at the comments and format`

File: README-en.md [hunk 3]
```diff
@@ -6,134 +6,141 @@
...
+* If the problem to fix already has a related issue, please assure that nobody submitted the corresponding PR. Then, please leave a comment about your fix plan.
+* If you want to submit several changes, please add the issue first. And describe your thoughts according to the actual situation(like difficulty and controversy)
+* A PR only contains one class of changes.
```
Comment (Felixnoo) [inline README-en.md:54]: `one class of changes` -> `changes of the same type`

File: README-en.md [hunk 4]
```diff
@@ -6,134 +6,141 @@
...
+* Add a new branch for each submit.
+* Avoid repeatedly closing and creating PRs for the same topic
+  * Avoid making frequent submit for the same PR, which will be great trouble for reviewers.
```
Comment (Felixnoo) [inline README-en.md:57]: `which will be great trouble for reviewers` -> `which makes it difficult for reviewers to review the PR`

File: README-en.md [hunk 5]
```diff
@@ -6,134 +6,141 @@
...
+  * If your PR is not already prepared for accepting review, please add prefix `WIP` before the title until you get readied.
```
Comment (Felixnoo) [inline README-en.md:58]: `get readied` -> `think the PR is ready for review`

File: README-en.md [hunk 6]
```diff
@@ -6,134 +6,141 @@
...
-## 其他参考
+In the process of PR review, there are usually three roles involved: AUTHOR, MAINTAINERS, and others.
+
+First of all, we need to figure out a question, why do we need a review? Review is critical, no matter it's code, documentation, or other types of files submitted in PR.
+
+* The expression of the most basic principles of open source(transparency and openness).
+  * The Review process is part of open source. It is not only the results that need to be opened but also the processes that need to be opened.
+  * Review does not mean an audit. Everyone can be a reviewer, and everyone will learn in this process.
+* Life cycle
+  * PR does not apply to urgent merge needs.
```
Comment (Felixnoo) [inline README-en.md:78]: `merge` -> `merging`

File: README-en.md [hunk 7]
```diff
@@ -6,134 +6,141 @@
...
+* Automated process
+  * Manage the review process with an automated tool like [Lighthouse](https://github.com/jenkins-x/lighthouse)
+  * Run as many automated tests as possible (unit tests, E2E tests, stress tests, etc.) before merging. 
+  * Avoid human intervention in automated processes
+  * If review is completed, but some manual verification is needed. To avoid an untimely automatic merge, you can block it with a comment command.
+    * For example, comment `/hold` to prevent robot-account auto-merging.
+
+## Community Operation      
+
+Different people have different meanings of the word **community**. The word community in this article is precisely the **open-source community**.
```
Comment (Felixnoo) [inline README-en.md:102]: `meanings` -> `understandings`

File: README-en.md [hunk 8]
```diff
@@ -6,134 +6,141 @@
...
+
+### SIG Operation
+
+Determine the admittance and approval system of SIG, and then control the number of SIG according to the team's actual situation. In this process, the quality of operations is the primary consideration.
+
+ The healthy development of SIG cannot be separated from openness, transparency, and diversification. There is no sense in existence for the formalistic and non-open-source SIG.
+
+### SIG Meeting
+
+* Meeting time need consider the needs of the majority of people
```
Comment (Felixnoo) [inline README-en.md:114]: `need consider` -> `should accommodate`

File: README-en.md [hunk 9]
```diff
@@ -6,134 +6,141 @@
...
+* Make sure meetings are regular
+  * In general, meetings once every two weeks are moderate.
+  * If the number of participants is relatively large and distributed in different time zones, two meetings can be divided based on participants' opinions.
+  * Notify the community as early as possible of canceling the meeting when no one available to chair the meeting can attend or in other cases where the meeting needs to be canceled.
+* The minutes of the meeting
+  * We can use [Google Document](https://docs.google.com/document/u/0/) for text records.
+  * We can use [YouTube](https://www.youtube.com/) for video records.
+
+## Others
+Here are some other recommended habits and methods:
```
Comment (Felixnoo) [inline README-en.md:136]: `habits and methods` -> `practices`

File: README-en.md [hunk 10]
```diff
@@ -6,134 +6,141 @@
...
+* Checking email habitually
```
Comment (Felixnoo) [inline README-en.md:137]: Check your email

File: README-en.md [hunk 11]
```diff
@@ -6,134 +6,141 @@
...
+  * Please be sure to associate your usual email address with GitHub, and then you can receive the notice information in issues and PR in time.
```
Comment (Felixnoo) [inline README-en.md:138]: Delete `be sure to`

**Review by Felixnoo — COMMENTED** [review]
Hi, here are some suggestions for your reference.

File: README-en.md [hunk 12]
```diff
@@ -6,134 +6,141 @@
...
+* If you want to submit several changes, please add the issue first. And describe your thoughts according to the actual situation(like difficulty and controversy)
+* A PR only contains one class of changes.
+* Add a new branch for each submit.
+* Avoid repeatedly closing and creating PRs for the same topic
+  * Avoid making frequent submit for the same PR, which will be great trouble for reviewers.
+  * If your PR is not already prepared for accepting review, please add prefix `WIP` before the title until you get readied.
+  * When making revisions based on reviewers' suggestions, avoid forced push `--force`, making reviewers unable to check your newest changing part.
+  * Keep your commit records as elegant as possible. In case there are too many reviews, the project owners will decide whether to squash your commit records at merge time.
+* Give as much detail as you can about your PR, like related issues(s), solving problems, and any context convenient to review.
+  * Provide screenshots of the effect before and after modification if it is related to UI.
```
Comment (AbdelouahabMbarki) [inline README-en.md:62]: `to UI` -> `to the UI`

File: README-en.md [hunk 13]
```diff
@@ -6,134 +6,141 @@
...
+  * List your self-test process as appropriate.
+  * Give your reason for the controversial part. 
+* If your PR hasn't been reviewed for more than a week, try to cc* related teams.
+  * If there is no related team to cc, you can find someone who has merged similar PR recently and explain that it is because there is no other way, and I apologize for bothering you.
+  * *cc stands for carbon copy which means that whose address appears after the cc: header would receive a copy of the message. Also, the Cc header would also appear inside the header of the received message.*
```
Comment (AbdelouahabMbarki) [inline README-en.md:67]: `inside the header` -> ` in the header`

File: README-en.md [hunk 14]
```diff
@@ -6,134 +6,141 @@
...
-还有一些其他的比较推荐的习惯、方法：
-* 习惯查看邮件
-  * 一定要把自己常用的邮箱地址关联到 GitHub 上，这样才能及时收到 issues 和 PR 中的互动信息
-* 通过电脑、手机来管理自己的日程
-  * 开源社区总会有各种各样的会议，如果没有使用日历的习惯的话，非常容易遗忘重要的事情
-
-## 其他参考
+In the process of PR review, there are usually three roles involved: AUTHOR, MAINTAINERS, and others.
+
+First of all, we need to figure out a question, why do we need a review? Review is critical, no matter it's code, documentation, or other types of files submitted in PR.
```
Comment (AbdelouahabMbarki) [inline README-en.md:72]: `Review is critical ` -> `The review is critical`

File: README-en.md [hunk 15]
```diff
@@ -6,134 +6,141 @@
...
+
+* The expression of the most basic principles of open source(transparency and openness).
+  * The Review process is part of open source. It is not only the results that need to be opened but also the processes that need to be opened.
```
Comment (AbdelouahabMbarki) [inline README-en.md:75]: `The Review process is part of open source. It is not only the results that need to be opened but also the processes that need to be opened.`
 i didnt't get the meaning behind this line

File: README-en.md [hunk 16]
```diff
@@ -6,134 +6,141 @@
...
+  * Review and merge within 2 to 7 days.
+* Politeness
+  * No one should review your PR, including maintainers.
+  * Thanks to each reviewer who contributed to your PR merge.
+  * Don't rush anyone to review your PR during the best review cycle.
+  * If you do need to ask for a review for help, please give instructions. And @ a team first, then directly @ a person, and apologize for interrupting.
+* Make a clear point.
+  * Try to avoid giving vague comments, and the writer needs to make changes based on the suggestions you give. 
+  * If you are not sure about the question, you can comment like this: "I have a feeling there may be a problem here," give the suggestion and why, and indicate that the comment does not block the PR merge.
+  * For the part in question, give information or data that supports your point. If relevant authoritative information is available, you can attach a related web link.
```
Comment (AbdelouahabMbarki) [inline README-en.md:91]: `information or data that supports` -> `information or data that support`

File: README-en.md [hunk 17]
```diff
@@ -6,134 +6,141 @@
...
+    * For example, code comments in PR are not standard, giving the official community documentation links.
+* Automated process
+  * Manage the review process with an automated tool like [Lighthouse](https://github.com/jenkins-x/lighthouse)
+  * Run as many automated tests as possible (unit tests, E2E tests, stress tests, etc.) before merging. 
+  * Avoid human intervention in automated processes
+  * If review is completed, but some manual verification is needed. To avoid an untimely automatic merge, you can block it with a comment command.
```
Comment (AbdelouahabMbarki) [inline README-en.md:97]: `If review is completed ` -> `If a review is completed `

File: README-en.md [hunk 18]
```diff
@@ -6,134 +6,141 @@
...
+
+## Community Operation      
+
+Different people have different meanings of the word **community**. The word community in this article is precisely the **open-source community**.
+
+There are many excellent practices and ways in community operation, such as social media, Meetup, and TOC.
+
+### SIG Operation
+
+Determine the admittance and approval system of SIG, and then control the number of SIG according to the team's actual situation. In this process, the quality of operations is the primary consideration.
```
Comment (AbdelouahabMbarki) [inline README-en.md:108]: `approval system of SIG` -> `and approval system of the SIG`

**Review by AbdelouahabMbarki — COMMENTED** [review]
I've left some comments here, please take a moment to have a look , well done 💯

File: README-en.md [hunk 19]
```diff
@@ -6,134 +6,141 @@
 ## Why?
 Why do I want to write the "Open-source Best Practice"?
 
-For thoese people who wants to try to participant in an open-source project, they usually feel confused about where get started. Hopefully, this practice can help more people. I believe that you can improve yourself no matter you're a normal developer or even not.
+Those who want to participate in an open-source project generally feel confused about where to get started. Hopefully, this practice can help more people. I believe that you can improve yourself no matter you're a normal developer or even not.
```
[the remark this answers is not in the document]
Comment (wjsvec) [reply README-en.md:9]: Thank you very much for your advice.

**Comment by wjsvec** [comment]
I make changes following the suggestions, thank you very much~😆

**Review by LinuxSuRen — APPROVED** [review]
hi @wjsvec , thanks so much for the translation work!

**Comment by LinuxSuRen** [comment]
hi @wjsvec , 抱歉打扰了，不知道你是否有兴趣一起长期维护这个”最佳实践”，有兴趣的话，可以加我微信（和 GitHub ID 一样），建群讨论。

**Comment by wjsvec** [comment]
> hi @wjsvec , 抱歉打扰了，不知道你是否有兴趣一起长期维护这个”最佳实践”，有兴趣的话，可以加我微信（和 GitHub ID 一样），建群讨论。

好的，乐意之极~

**Pull request merged**